### ISSUE TITLE
Money pattern

### ISSUE THREAD

**Issue by iluwatar** [iluwatar/java-design-patterns#1305]
### Description
The Money design pattern, as described by Martin Fowler, provides a way to encapsulate monetary values and their associated currencies. This pattern helps in handling calculations and operations involving money in a clear and reliable manner, avoiding common issues like rounding errors and ensuring currency consistency.

Key Elements of the Money Pattern:
- Encapsulation of a monetary value along with its currency.
- Operations involving monetary values, such as addition, subtraction, and multiplication, which consider the currency.
- Proper handling of rounding issues.
- Conversion between different currencies using an exchange rate.

### References
- [Martin Fowler's Money Pattern](https://martinfowler.com/eaaCatalog/money.html)
- [Patterns of Enterprise Application Architecture by Martin Fowler (Page 303)](https://books.google.fi/books?id=vqTfNFDzzdIC&pg=PA303#v=onepage&q&f=false)

### Acceptance Criteria
1. Create a `Money` class that encapsulates a monetary value and its associated currency.
2. Implement operations (addition, subtraction, multiplication) on the `Money` class that handle currency consistency and rounding issues.
3. Add functionality for currency conversion with appropriate handling of exchange rates.

**Comment by sophonie** [comment]
Hi,
I am very enthusiasm to lean on that issue. I read the topic of this pattern but what is the purpose of this issue.
Would I have to code this pattern in Java ? I don't see your request.
Please I would like to have more informations about your aim.

Best regards.
Sophonie.

**Comment by iluwatar** [comment]
Yes @sophonie, I'm sorry for the poor description in this issue. The aim is to create an example implementation of Money pattern in this repository. You can check how the other patterns have been implemented and follow the same style. There are also brief contribution instruction at https://github.com/iluwatar/java-design-patterns/wiki/01.-How-to-contribute

Please let me know if you want to try implementing this pattern.

**Comment by ToxicDreamz** [comment]
I've submitted a PR for this issue. See -> #1673

**Comment by iluwatar** [comment]
This issue is free for taking again.

**Comment by Liu-rj** [comment]
Hi, I'd like to work on this issue. Could it be assigned to me if still available?

**Comment by iluwatar** [comment]
Yes @Liu-rj, please go ahead

**Comment by iluwatar** [comment]
@Liu-rj are you working on this?

**Comment by Liu-rj** [comment]
@iluwatar yeah, I will fix that pr #1984 these days once I have time.

**Comment by annieS987** [comment]
@iluwatar Hi, can I work on this one as well?

**Comment by Ahmed-Taha-981** [comment]
Hi, this issue seems interesting, I'll start on my implementation, could you please assign it to me.

**Comment by PlumHeadd** [comment]
Hi @iluwatar, please assign this issue to me so I can implement this pattern. Thank you.
